api/access/server.go

```diff
@@ -0,0 +1,515 @@
+// for data retrieval.
+func NewServer(index dps.Reader, codec dps.Codec, invoker Invoker) *Server {
+	s := Server{
+		index:   index,
+		codec:   codec,
+		invoker: invoker,
+	}
+
+	return &s
+}
+
+func (s *Server) Ping(_ context.Context, _ *access.PingRequest) (*access.PingResponse, error) {
+	return &access.PingResponse{}, nil
+}
+
+func (s *Server) GetLatestBlockHeader(ctx context.Context, _ *access.GetLatestBlockHeaderRequest) (*access.BlockHeaderResponse, error) {
+	height, err := s.index.Last()
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve last block height: %w", err)
+	}
+
+	req := access.GetBlockHeaderByHeightRequest{
+		Height: height,
+	}
+
+	return s.GetBlockHeaderByHeight(ctx, &req)
+}
+
+func (s *Server) GetBlockHeaderByID(ctx context.Context, in *access.GetBlockHeaderByIDRequest) (*access.BlockHeaderResponse, error) {
+	blockID := flow.HashToID(in.Id)
+	height, err := s.index.HeightForBlock(blockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get height for block %x: %w", blockID, err)
+	}
+
+	req := access.GetBlockHeaderByHeightRequest{
+		Height: height,
+	}
+
+	return s.GetBlockHeaderByHeight(ctx, &req)
+}
+
+func (s *Server) GetBlockHeaderByHeight(_ context.Context, in *access.GetBlockHeaderByHeightRequest) (*access.BlockHeaderResponse, error) {
+	header, err := s.index.Header(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve block header at height %d: %w", in.Height, err)
+	}
+
+	blockMsg, err := convert.BlockHeaderToMessage(header)
+	if err != nil {
+		return nil, fmt.Errorf("could not convert block header to RPC entity: %w", err)
+	}
+
+	resp := access.BlockHeaderResponse{
+		Block: blockMsg,
+	}
+
+	return &resp, err
+}
+
+func (s *Server) GetLatestBlock(ctx context.Context, in *access.GetLatestBlockRequest) (*access.BlockResponse, error) {
+	height, err := s.index.Last()
+	if err != nil {
+		return nil, fmt.Errorf("could not get last height: %w", err)
+	}
+
+	req := &access.GetBlockByHeightRequest{
+		Height: height,
+	}
+
+	return s.GetBlockByHeight(ctx, req)
+}
+
+func (s *Server) GetBlockByID(ctx context.Context, in *access.GetBlockByIDRequest) (*access.BlockResponse, error) {
+	blockID := flow.HashToID(in.Id)
+	height, err := s.index.HeightForBlock(blockID)
+	if err != nil {
+		return nil, fmt.Errorf("could not get height for block %x: %w", blockID, err)
+	}
+
+	req := access.GetBlockByHeightRequest{
+		Height: height,
+	}
+
+	return s.GetBlockByHeight(ctx, &req)
+}
+
+func (s *Server) GetBlockByHeight(_ context.Context, in *access.GetBlockByHeightRequest) (*access.BlockResponse, error) {
+	header, err := s.index.Header(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get header for height %d: %w", in.Height, err)
+	}
+
+	sealIDs, err := s.index.SealsByHeight(in.Height)
+	if err != nil {
+		return nil, fmt.Errorf("could not get seals for height %d: %w", in.Height, err)
+	}
+
+	seals := make([]*entities.BlockSeal, 0, len(sealIDs))
```

Comment:
In analogy with other functions that are simpler, this could be `sealMsgs`, and `entity` in the loop could then be `sealMsg`, so it would be a consistent append (`sealMsgs = append(sealMsgs, sealMsg)`).